### PR TITLE
[Android] result in RCTAsyncStorage.multiGet could be null.

### PR DESCRIPTION
**Pull request by zongjingyao**
App will crash if result is null.

### CODE REVIEW

File: Libraries/Storage/AsyncStorage.js [hunk 1]
```diff
@@ -264,7 +264,7 @@ var AsyncStorage = {
       // Is there a way to avoid using the map but fix the bug in this breaking test?
       // https://github.com/facebook/react-native/commit/8dd8ad76579d7feef34c014d387bf02065692264
       const map = {};
-      result.forEach(([key, value]) => map[key] = value);
+      result && result.forEach(([key, value]) => map[key] = value);
```
Comment (geirman) [inline Libraries/Storage/AsyncStorage.js]: I don't understand this complaint 🤔

**Comment by geirman** [comment]
@facebook-github-bot ship it

File: Libraries/Storage/AsyncStorage.js (hunk 1 above)
Comment (ericvicenti) [inline Libraries/Storage/AsyncStorage.js]: To make this warning go away:

```
result && result.forEach(([key, value]) => { map[key] = value; return value; });
```

Personally I don't mind, but it apparently doesn't want you to assign in the return line.

Comment (geirman) [inline Libraries/Storage/AsyncStorage.js]: Ah, that makes sense! 

@zongjingyao will you please update so that it passes lint?

Comment (zongjingyao) [inline Libraries/Storage/AsyncStorage.js]: Updated :)

**Comment by lacker** [comment]
I think the test failure is spurious here - let me try rerunning circle and if that looks good I will merge it.

**Comment by lacker** [comment]
Rebasing - I think test failures were just due to oldness, despite the one-line nature of this fix.

**Pull request closed without merging**